omod-dstu3/src/main/java/org/openmrs/module/fhir2/providers/DiagnosticReportDstu3FhirResourceProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.providers;
+
+import javax.validation.constraints.NotNull;
+
+import ca.uhn.fhir.rest.annotation.*;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.convertors.VersionConvertor_30_40;
+import org.hl7.fhir.dstu3.model.DiagnosticReport;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.openmrs.module.fhir2.api.FhirDiagnosticReportService;
+import org.openmrs.module.fhir2.util.FhirServerUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Qualifier("fhirDstu3Resources")
+@Setter(AccessLevel.PACKAGE)
+public class DiagnosticReportDstu3FhirResourceProvider implements IResourceProvider {
+	
+	@Autowired
+	private FhirDiagnosticReportService diagnosticReportService;
+	
+	@Override
+	public Class<? extends IBaseResource> getResourceType() {
+		return DiagnosticReport.class;
+	}
+	
+	@Read
+	@SuppressWarnings("unused")
+	public DiagnosticReport getDiagnosticReportById(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.DiagnosticReport diagnosticReport = diagnosticReportService
+		        .getDiagnosticReportByUuid(id.getIdPart());
+		if (diagnosticReport == null) {
+			throw new ResourceNotFoundException("Could not find diagnosticReport with Id " + id.getIdPart());
+		}
+		
+		return VersionConvertor_30_40.convertDiagnosticReport(diagnosticReport);
+	}
+	
+	// TODO: Required or not?
+	@Create
+	public MethodOutcome createDiagnosticReport(@ResourceParam org.hl7.fhir.r4.model.DiagnosticReport diagnosticReport) {
```

Comment:
Required, but we will obviously be given the R3 version of the DiagnosticReport here.